File: src/utils/console.ts
// Third-Party libraries
import chalk from 'chalk'

// Constants
import type { ERROR_MESSAGES } from '@/constants/errors'

// Types
import type { MessageType } from '@/types/message'

interface Props {
  type: MessageType
  message: typeof ERROR_MESSAGES | string
}

export function writeMessage({ type, message }: Props) {
  if (type === 'success') console.log(chalk.green(message))
  if (type === 'info') console.log(chalk.blue(message))
  if (type === 'error') console.log(chalk.red(message))
  if (type === 'config') console.log(chalk.white(message))
}

File: src/utils/console.rs
use crate::constants::{DEFAULT_COLOR, MESSAGE_COLORS};
use crate::enums;

pub struct WriteMessageProps {
    pub message_type: enums::MessageType,
    pub message: &'static str,
}

pub fn write_message(props: WriteMessageProps) {
    let default_color_string = DEFAULT_COLOR.to_string();

    let color = MESSAGE_COLORS
        .get(&props.message_type)
        .unwrap_or(&default_color_string);

    println!("{}{}{}", color, props.message, DEFAULT_COLOR);
}
